concurrency-limits-core/src/main/java/com/netflix/concurrency/limits/limit/window/PercentileSampleWindowFactory.java

```diff
@@ -19,18 +19,20 @@
 
 public class PercentileSampleWindowFactory implements SampleWindowFactory {
     private final double percentile;
+    private final int windowSize;
 
-    private PercentileSampleWindowFactory(double percentile) {
+    private PercentileSampleWindowFactory(double percentile, int windowSize) {
         this.percentile = percentile;
+        this.windowSize = windowSize;
     }
 
-    public static PercentileSampleWindowFactory of(double percentile) {
+    public static PercentileSampleWindowFactory of(double percentile, int windowSize) {
         Preconditions.checkArgument(percentile > 0 && percentile < 1.0, "Percentile should belong to (0, 1.0)");
-        return new PercentileSampleWindowFactory(percentile);
+        return new PercentileSampleWindowFactory(percentile, windowSize);
     }
 
     @Override
     public ImmutablePercentileSampleWindow newInstance() {
-        return new ImmutablePercentileSampleWindow(percentile);
+        return new ImmutablePercentileSampleWindow(percentile, (int) (windowSize * 1.2));
```

Comment:
what's the `* 1.2` for?

Reply:
that was introduced as a heuristic value for case, when under very heavy load we might insert next sample, while new sampleWindow was not created yet. Doesn't make sense if we're making this https://github.com/Netflix/concurrency-limits/pull/119#discussion_r284789865